### PR TITLE
SLCORE-612 Move taint vulnerabilities management to the backend

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: rpc-impl/src/main/java/org/sonarsource/sonarlint/core/rpc/impl/SonarLintRpcServerImpl.java [hunk 1]
```diff
@@ -185,6 +192,11 @@ public NewCodeRpcService getNewCodeService() {
     return new NewCodeRpcServiceDelegate(this);
   }
 
+  @Override
+  public TaintVulnerabilityTrackingRpcService getTaintVulnerabilityService() {
```
Comment (henryju) [inline rpc-impl/src/main/java/org/sonarsource/sonarlint/core/rpc/impl/SonarLintRpcServerImpl.java:196]:
```suggestion
  public TaintVulnerabilityTrackingRpcService getTaintVulnerabilityTrackingService() {
```
for consistency with the type?

**Review by henryju — APPROVED** [review]
LGTM!

File: server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/XodusServerIssueStore.java [hunk 2]
```diff
@@ -957,11 +948,15 @@ private static Location toProtoLocation(ServerIssueLocation l) {
...
-  static void checkCurrentSchemaVersion(StoreTransaction txn) {
+  static void migrate(StoreTransaction txn) {
     var currentSchemaVersion = getCurrentSchemaVersion(txn);
-    if (currentSchemaVersion < CURRENT_SCHEMA_VERSION) {
+    if (currentSchemaVersion < 1) {
       // Migrate v0 to v1: force re-sync of taint vulnerabilities
       txn.getAll(BRANCH_ENTITY_TYPE).forEach(b -> b.setProperty(LAST_TAINT_SYNC_PROPERTY_NAME, Instant.EPOCH));
+    }
+    if (currentSchemaVersion < CURRENT_SCHEMA_VERSION) {
+      // Migrate v1 to v2: attribute a UUID to each taint
```
[the remark this answers is not in the document]
Comment (damien-urruty-sonarsource) [reply server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/XodusServerIssueStore.java:958]: Could you explain why "attribute" wouldn't work here?

[the remark this answers is not in the document]
Comment (henryju) [reply server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/XodusServerIssueStore.java:958]: I don't think the "attribute" verb has the same meaning that "attribuer" in French. But I could be wrong.

Maybe simply "set"?

[the remark this answers is not in the document]
Comment (damien-urruty-sonarsource) [reply server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/XodusServerIssueStore.java:958]: I will use "assign"

**Pull request merged**